dashboard/src/pages/[workspaceSlug]/[appSlug]/users/index.tsx

```diff
@@ -1,38 +1,379 @@
+              _ilike: `%${searchString}%`,
+            },
+          },
+          {
+            email: {
+              _ilike: `%${searchString}%`,
+            },
+          },
+        ],
+      },
+      limit: limit.current,
+      offset: offset * limit.current,
+    },
+    client: remoteProjectGQLClient,
+  });
+
+  /**
+   * We want to update the number of pages when the data changes
+   * (either fetch for the first time or making a search).
+   */
+  useEffect(() => {
+    if (loadingRemoteAppUsersQuery) {
+      return;
+    }
+
+    setNrOfPages(
+      Math.ceil(
+        dataRemoteAppUsers.usersAggregate.aggregate.count / limit.current,
+      ),
+    );
+  }, [dataRemoteAppUsers, loadingRemoteAppUsersQuery]);
+
+  const handleSearchStringChange = useMemo(
+    () =>
+      debounce((event: ChangeEvent<HTMLInputElement>) => {
+        setCurrentPage(1);
+        setSearchString(event.target.value);
+      }, 500),
+    [],
+  );
+
+  useEffect(
+    () => () => handleSearchStringChange.cancel(),
+    [handleSearchStringChange],
+  );
+
+  function handleCreateUser() {
+    const signUpUrl = `${generateAppServiceUrl(
+      currentApplication?.subdomain,
+      currentApplication?.region.awsName,
+      'auth',
+    )}/v1/signup/email-password`;
+
+    openDialog('CREATE_USER', {
+      title: 'Create User',
+      payload: {
+        onSubmit: async ({ email, password }) => {
+          await axios.post(signUpUrl, {
+            email,
+            password,
+          });
+          await refetchProjectUsers();
+          closeDialog();
+        },
+      },
+      props: {
+        titleProps: { className: 'mx-auto' },
+        PaperProps: { className: 'max-w-md' },
+      },
+    });
   }
 
-  return (
-    <NhostApolloProvider
-      graphqlUrl={`${generateAppServiceUrl(
-        currentApplication.subdomain,
-        currentApplication.region.awsName,
-        'graphql',
-      )}/v1`}
-      fetchPolicy="cache-first"
-      headers={{
-        'x-hasura-admin-secret':
-          process.env.NEXT_PUBLIC_ENV === 'dev'
-            ? 'nhost-admin-secret'
-            : currentApplication.hasuraGraphqlAdminSecret,
-      }}
-    >
-      <Container>
-        <UsersList />
+  /**
+   * This will change the `disabled` field in the user to its opposite.
+   */
+  async function handleBanUser(user: RemoteAppUser) {
+    const banUser = updateUser({
+      variables: {
+        id: user.id,
+        user: {
+          disabled: !user.disabled,
+        },
+      },
+    });
+
+    await toast.promise(
+      banUser,
+      {
+        loading: user.disabled ? 'Unbanning user...' : 'Banning user...',
+        success: user.disabled
+          ? 'User unbanned succesfully.'
+          : 'User banned succesfully',
+        error: user.disabled
+          ? 'Error trying to unban user.'
+          : 'Error trying to ban user.',
+      },
+      { ...toastStyleProps },
+    );
+    await refetchProjectUsers();
```

Comment:
Unfortunately, we cannot easily do it the way we're doing it in the other pages like `Roles & Permissions` because here we are not fetching from our remote but from the user's app (second Apollo GQL client) and also using a \`dialog\` which lives outside the dom nodes from this, I tried this approach initially since it would greatly improve the code but when deleting /updating the user, the Apollo client cannot find the given query \`getRemoteUsersWhere\` this https://github.com/apollographql/apollo-client/issues/5419#issuecomment-598065442 is a good comment about the problem

If we import the query itself, we need to give it the variables for which it was previously run (passing down the props) if not, running the query as is, we would see inconsistent behavior when a user searchs for a user "sam," deletes it, and the query refetchs the initial page.

We could just update the cache directly after mutating a user, but I'm not so sure we all want that...

```typescript
 /**
  * Function to be called after a successful action.
  *
  * @example onSuccessfulAction={() => refetch()}
  * @example onSuccessfulAction={() => router.reload()}
  */
 onSuccessfulAction?: () => Promise<void> | void | Promise<T>;
```

I think something similar to your comment about `onSubmit` work best.